### PR TITLE
WIP global: admin modelviews registration

### PR DESCRIPTION
**Pull request by krzysztof**
* Added model views registration through entry points

### CODE REVIEW

**Comment by krzysztof** [comment]
This PR provides relevant admin registration code: https://github.com/inveniosoftware/invenio-pidstore/pull/30

File: invenio_admin/config.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Admin config."""
+
+from __future__ import unicode_literals
```
Comment (jirikuncar) [inline invenio_admin/config.py]: is it really wanted?

File: invenio_admin/ext.py [hunk 2]
```diff
@@ -26,25 +26,38 @@
...
+        register_call(self._admin)
+
+    def init_app(self, app, entrypoint_name='invenio_admin.register',
+                 **kwargs):
         """Flask application initialization."""
         self.init_config(app)
         app.register_blueprint(blueprint)
         app.extensions['invenio-admin'] = self
+        admin = Admin(app, name=app.config['ADMIN_APPNAME'],
+                      template_mode='bootstrap3')
```
Comment (jirikuncar) [inline invenio_admin/ext.py]: `template_mode=kwargs.get('template_mode', 'bootstrap3')`

File: invenio_admin/ext.py [hunk 3]
```diff
@@ -26,25 +26,38 @@
...
 
 class InvenioAdmin(object):
     """Invenio-Admin extension."""
 
-    def __init__(self, app=None):
+    def __init__(self, app=None, **kwargs):
         """Extension initialization."""
-        _('A translation string')
         if app:
             self.init_app(app)
```
Comment (jirikuncar) [inline invenio_admin/ext.py]: shall we pass `kwargs` to `init_app` too?

File: invenio_admin/ext.py [hunk 4]
```diff
@@ -26,25 +26,38 @@
...
         """Flask application initialization."""
         self.init_config(app)
         app.register_blueprint(blueprint)
         app.extensions['invenio-admin'] = self
+        admin = Admin(app, name=app.config['ADMIN_APPNAME'],
+                      template_mode='bootstrap3')
+        self._admin = admin
+        if entrypoint_name:
+            for ep in pkg_resources.iter_entry_points(
+                    group='invenio_admin.register'):
```
Comment (jirikuncar) [inline invenio_admin/ext.py]: `invenio_admin.view` or `invenio_admin.app(s)`?

File: invenio_admin/config.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_admin/config.py]: ping @krzysztof

Comment (krzysztof) [inline invenio_admin/config.py]: Perhaps the ``u`` was unnecessary two lines below: https://github.com/inveniosoftware/invenio-admin/pull/2/files#diff-3fca13c63e83fb9792613cf0ea118207R24

As for ``unicode_literals`` I guess it is, given that we still support Python2 ?

Comment (jirikuncar) [inline invenio_admin/config.py]: `unicode_literals` is for *lazy* developers. Normally you don't need it ... more details IRL.

File: setup.py [hunk 5]
```diff
@@ -61,6 +61,8 @@
 
 install_requires = [
     'Flask-BabelEx>=0.9.2',
+    'Flask-Admin>=1.3.0',
```
Comment (jirikuncar) [inline setup.py]: please sort the requirements

File: invenio_admin/ext.py [hunk 6]
```diff
@@ -26,25 +26,39 @@
...
+        """Register the admin views within register_call to current admin."""
+        register_call(self._admin)
+
+    def init_app(self, app, entrypoint_name='invenio_admin.views',
+                 **kwargs):
         """Flask application initialization."""
         self.init_config(app)
         app.register_blueprint(blueprint)
         app.extensions['invenio-admin'] = self
+        admin = Admin(app, name=app.config['ADMIN_APPNAME'],
```
Comment (jirikuncar) [inline invenio_admin/ext.py]: IMHO `self.admin` is fine. Is there a reason why do you want to *protect* the `Admin` instance?

Comment (krzysztof) [inline invenio_admin/ext.py]: No good reason. Changing it to ``self.admin``.

Comment (jirikuncar) [inline invenio_admin/ext.py]: :+1:

File: invenio_admin/config.py (hunk 1 above)
Comment (krzysztof) [inline invenio_admin/config.py]: :+1:

File: invenio_admin/views.py [hunk 7]
```diff
@@ -27,8 +27,33 @@
...
+
+    def __init__(self, *args, **kwargs):
+        """Constructor for the AdminIndexView."""
+        super(ProtectedAdminIndexView, self).__init__(*args, **kwargs)
+
+    @expose('/')
+    def index(self):
+        """Index view."""
+        if not current_user.is_authenticated():
+            return False
```
Comment (jirikuncar) [inline invenio_admin/views.py]: `return False` or `abort(401)`?

File: invenio_admin/ext.py [hunk 8]
```diff
@@ -26,25 +26,39 @@
...
+                      template_mode=kwargs.get('template_mode', 'bootstrap3'),
+                      index_view=ProtectedAdminIndexView(),
+                      )
+        self.admin = admin
```
Comment (jirikuncar) [inline invenio_admin/ext.py]: you can join it with line 54 as `admin` is not used anywhere except assignment to `self.admin`.

File: tests/conftest.py [hunk 9]
```diff
@@ -27,15 +27,63 @@
...
     app.config.update(
-        TESTING=True
+        TESTING=True,
+        SECRET_KEY="SECRET_KEY",
     )
+
+    runner = CliRunner()
+    script_info = ScriptInfo(create_app=lambda info: app)
+
+    runner.invoke(db_cmd, ['init'], obj=script_info)
```
Comment (lnielsen) [inline tests/conftest.py]:
```
assert runner.invoke(db_cmd, ['init'], obj=script_info).result_code == 0
```

File: .travis.yml [hunk 10]
```diff
@@ -33,10 +33,20 @@ language: python
...
 
+services:
+  - mysql
+  - postgresql
+
 env:
-  - REQUIREMENTS=lowest
-  - REQUIREMENTS=release
-  - REQUIREMENTS=devel
+  - REQUIREMENTS=lowest SQLALCHEMY_DATABASE_URI="sqlite:///test.db"
```
Comment (lnielsen) [inline .travis.yml]: Just test with PostgreSQL

Comment (krzysztof) [inline .travis.yml]: Done.

File: tests/conftest.py (hunk 9 above)
Comment (krzysztof) [inline tests/conftest.py]: Replaced this with create/drop from ``sqlalchemy_utils.functions``

**Comment by lnielsen** [comment]
Seems like tests are failing due to isort. Just comment it out in run-tests.sh

**Comment by krzysztof** [comment]
That's interesting as the commit that made these fail was: https://github.com/krzysztof/invenio-admin/commit/9c0b1116912a896fd0baf12b4630db8386f92399

And the failing tests are only development tests, which passed before when invenio-db was not in the development dependencies..

**Comment by lnielsen** [comment]
Merged via 28f521521bd74673dfbdcb80a73ce65145a17501

**Pull request closed without merging**